src_c/display.c

```diff
@@ -879,18 +885,25 @@ pg_set_mode(PyObject *self, PyObject *arg, PyObject *kwds)
                 }
             }
 
-            if (flags & PGS_SCALED && !(flags & PGS_FULLSCREEN)) {
-                SDL_DisplayMode dm;
-                int xscale, yscale;
-                if (SDL_GetDesktopDisplayMode(display, &dm) != 0) {
-                    return RAISE(pgExc_SDLError, SDL_GetError());
+            if (SDL_GetDesktopDisplayMode(display, &dm) != 0) {
+                return RAISE(pgExc_SDLError, SDL_GetError());
+            }
+
+            if (flags & PGS_SCALED){
+                if (!(flags & PGS_FULLSCREEN)) {
+                    int xscale, yscale;
+
+                    xscale = dm.w / w;
```

Comment:
Is it true that here `w` and `h` will never be zero?

Reply:
Right now, they can't be zero. Should we add a check now for the sake of future-proofing?